packages/beacon-node/src/chain/interface.ts

```diff
@@ -95,7 +95,7 @@ export interface IBeaconChain {
   readonly checkpointBalancesCache: CheckpointBalancesCache;
   readonly producedBlobSidecarsCache: Map<BlockHash, {blobSidecars: deneb.BlobSidecars; slot: Slot}>;
   readonly producedBlindedBlobSidecarsCache: Map<BlockHash, {blobSidecars: deneb.BlindedBlobSidecars; slot: Slot}>;
-  readonly producedBlockRoot: Set<RootHex>;
+  readonly producedBlockRoot: Map<RootHex, allForks.ExecutionPayload | null>;
```

Comment:
Add metrics to track the size of this cache. Also for others related to block production added for deneb